### PR TITLE
Use --error-color instead of a fixed color

### PR DESCRIPTION
**Pull request by KTibow**
## Proposed change

This makes the error-card use `--error-color` as the background color, instead of a set one, so you can theme it.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
frontend:
  themes:
    errorblues:
      error-color: blue
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-error-card.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class HuiErrorCard extends LitElement implements LovelaceCard {
     return css`
       :host {
         display: block;
-        background-color: #ef5350;
+        background-color: var(--error-color);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-error-card.ts:46]: Is this the same color?

Comment (thomasloven) [reply src/panels/lovelace/cards/hui-error-card.ts:46]: Default is `#E45E65`
image

Comment (KTibow) [reply src/panels/lovelace/cards/hui-error-card.ts:46]: Should it be updated in the place where error color is defined, or is this a bad idea?

Comment (thomasloven) [reply src/panels/lovelace/cards/hui-error-card.ts:46]: IIRC the text color is white on the error card, that might be a bit low contrast with the new color, but I'm not sure what I think. I like there being fewer colors and more variables...
image

File: src/panels/lovelace/cards/hui-error-card.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ export class HuiErrorCard extends LitElement implements LovelaceCard {
...
         color: white;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-error-card.ts:47]: If we make the background themeable, the font color should also be themeable... Something like:
`color: var(--color-on-error, white)`

File: src/panels/lovelace/cards/hui-error-card.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-error-card.ts:46]: I think it is ok... I would like to have just 1 error color

File: src/panels/lovelace/cards/hui-error-card.ts (hunk 2 above)
Comment (KTibow) [reply src/panels/lovelace/cards/hui-error-card.ts:47]:
```suggestion
        color: var(--error-card-text-color);
```

File: src/panels/lovelace/cards/hui-error-card.ts [hunk 3]
```diff
@@ -43,8 +43,8 @@ export class HuiErrorCard extends LitElement implements LovelaceCard {
     return css`
       :host {
         display: block;
-        background-color: #ef5350;
-        color: white;
+        background-color: var(--error-color);
+        color: var(--error-card-text-color, white);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-error-card.ts:47]: This is too specific, we want to use it everywhere we use the error color as the background.

Comment (KTibow) [reply src/panels/lovelace/cards/hui-error-card.ts:47]: Where's that? I don't see anywhere else.

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-error-card.ts:47]: Future proof :-)

Comment (zsarnett) [reply src/panels/lovelace/cards/hui-error-card.ts:47]: regardless, we should limit ourselves to be only on  a card

Comment (KTibow) [reply src/panels/lovelace/cards/hui-error-card.ts:47]: So what's a good name? `error-text-color`? In some places `error-color` is used as the color of text.

Comment (ludeeus) [reply src/panels/lovelace/cards/hui-error-card.ts:47]: https://github.com/home-assistant/frontend/pull/6961#discussion_r487569117

Comment (KTibow) [reply src/panels/lovelace/cards/hui-error-card.ts:47]:
```suggestion
        color: var(--on-error-color, white);
```

File: src/panels/lovelace/cards/hui-error-card.ts [hunk 4]
```diff
@@ -43,8 +43,8 @@ export class HuiErrorCard extends LitElement implements LovelaceCard {
     return css`
       :host {
         display: block;
-        background-color: #ef5350;
-        color: white;
+        background-color: var(--error-color);
+        color: var(--on-error-color, white);
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-error-card.ts:47]:
```suggestion
        color: var(--color-on-error, white);
```

Let's not be stubborn and just use Bram's suggestion. As it follows the Material Variable naming 

Ex. var(--mdc-theme-on-primary)

**Review by zsarnett — APPROVED** [review]

**Pull request merged**